src/cli.rs

```diff
@@ -78,25 +90,68 @@ impl Cli {
             "Starting Quilkin"
         );
 
+        let config = Arc::new(Self::read_config(self.config)?);
+        let _admin_task = if self.no_admin {
+            config.admin.remove();
+            None
+        } else {
+            if let Some(address) = self.admin_address {
+                config
+                    .admin
+                    .store(Arc::new(crate::config::Admin { address }));
+            }
+            Some(tokio::spawn(crate::admin::server(config.clone())))
```

Comment:
Pretty sure this is the offending line causing the double creation of the admin server, since both `Proxy::run` and `ControlPlane::from_arc` both create admin instances and run them.

Reply:
Doing some debugging ([code](https://github.com/markmandel/quilkin/commit/09de12a7f906e9ea0451689d54b7ff0a40fdd8fc)), it looks like the `Config` that comes from this line:

https://github.com/markmandel/quilkin/commit/09de12a7f906e9ea0451689d54b7ff0a40fdd8fc#diff-b2812f19576dd53d0c35b107a322f58a00fce6977f4b5976e1961853982af3ccR93

```rust
let config = Arc::new(Self::read_config(self.config)?);
```

Has a `None` admin field, but looking at the code I'm not sure why it's `None` when it should have the default admin port 🤔 

You can see it in my debug logs:

```log
{"timestamp":"2022-09-22T04:41:50.829061Z","level":"DEBUG","fields":{"message":"Main Config","no_admin":false,"config":"Config { admin: None, clusters: None, filters: None, management_servers: None, proxy: None, version: None, metrics: Metrics { active_clusters: GenericGauge { v: Value { desc: Desc { fq_name: \"cluster_active\", help: \"Number of currently active clusters.\", const_label_pairs: [], variable_labels: [], id: 6550223101314893, dim_hash: 9568000379080501117 }, val: AtomicI64 { inner: 0 }, val_type: Gauge, label_pairs: [] } }, active_endpoints: GenericGauge { v: Value { desc: Desc { fq_name: \"cluster_active_endpoints\", help: \"Number of currently active endpoints.\", const_label_pairs: [], variable_labels: [], id: 9175583387471564798, dim_hash: 12641526325508385022 }, val: AtomicI64 { inner: 0 }, val_type: Gauge, label_pairs: [] } } } }"},"target":"quilkin::cli"}
```

I'll keep digging into this tomorrow unless you get to it first.

Reply:
I think I've found the issue. I'll send a PR to your PR today.